semeio/workflows/ahm_analysis/ahmanalysis.py

```diff
@@ -423,20 +394,23 @@ def calc_delta_grid(
     return mygrid_ok_short
 
 
+def _get_parameter_keys(parameters, prior_keys):
+    transfp_keys = []
+    for parameter in parameters:
+        for key in prior_keys:
+            if "_" + parameter + ":" in key:
+                transfp_keys.append(key)
+    return set(prior_keys).difference(transfp_keys)
+
+
 def get_updated_parameters(prior_data, parameters):
     """make list of updated parameters
     (excluding duplicate transformed parameters)
     """
-    transfp_keys = []
-    p_keys = []
-    p_keysf = []
-    for paramet in parameters:
-        for key in prior_data.keys():
-            if "_" + paramet + ":" in key:
-                transfp_keys.append(key)
-    p_keys = set(prior_data.keys()).difference(transfp_keys)
+    parameter_keys = _get_parameter_keys(parameters, prior_data.keys())
     # remove parameters with constant prior distribution
-    for dkey in p_keys:
+    p_keysf = []
+    for dkey in parameter_keys:
         if prior_data[dkey].ndim > 1:
             print("WARNING: Parameter " + dkey + " defined several times.")
```

Comment:
I am a bit confused about this, what is this meant to do? It is never touched by a test, so a bit hard to determine.

Reply:
this is checking if some parameters are defined several times (mistakenly introduced by the user). If there are constant parameters defined several times ert will not crash and most likely do not give a message (so in this case it is just to give a warning (make the user aware of it) since it will not affect our script results) but if they are not constant ert will crash (so this is not an issue ).

Reply:
Thanks for the clarification, could you add a test that touches this part of the code? Like you said, it does not affect our script, but if it happens because of a user error we should perhaps throw an error? It should of course be the responsibility of `ert` to validate the input, and we should not have to do that at this point ideally. So perhaps the correct course of action is to create an issue in `ert`?

Reply:
I have added a test.